Modify that i missed

JungDaewook opened the pull request.

Commit:
modify by code review

Commit:
modify that I missed.
1. limits of todoinputText.(255)
2. limits of the amount of todoinputText on the todoList.(20)

Commit:
finish the fifth part DeleteTodoItem.
made a new function named removeTodoItem
modified js file's name to functionsOfTodoList because it has not only function addTodoItem but also other functions of this project

unpo88 commented on Todo_List.html:
```diff
@@ -4,11 +4,17 @@
     <input type = 'button' value = '추가' onclick = 'addTodoItem()' />
-    <ul id = 'todoList'></ul>
+    <ul id = 'todoList'>
+        <style>
+            li:nth-of-type(n+20){
+                display: none;
+            }
+        </style>
```
@JungDaewook 

해당 스타일은 아래 요구사항을 만족시키기 위해 작성된 코드로 보이는데요 🤔 

- “할 일 목록"을 보여줄 수 있는 **리스트**가 존재한다.
    - 할 일 목록은 최대 20개까지 입력 가능하다.

하지만, 해당 코드는 최대 20개 까지 입력이 가능하다가 아니라

"20개 이후의 리스트 항목을 숨김처리(`display: none`)한다"의 요구사항을 만족시키는 것 같아요.

실제로는, 20개 이후의 항목들이 생성되고 있습니다. 수정이 필요하지 않을까요?

JungDaewook replied:
> @JungDaewook
> 
> 해당 스타일은 아래 요구사항을 만족시키기 위해 작성된 코드로 보이는데요 🤔
> 
> * “할 일 목록"을 보여줄 수 있는 **리스트**가 존재한다.
>   
>   * 할 일 목록은 최대 20개까지 입력 가능하다.
> 
> 하지만, 해당 코드는 최대 20개 까지 입력이 가능하다가 아니라
> 
> "20개 이후의 리스트 항목을 숨김처리(`display: none`)한다"의 요구사항을 만족시키는 것 같아요.
> 
> 실제로는, 20개 이후의 항목들이 생성되고 있습니다. 수정이 필요하지 않을까요?

[답변/수정 완료]
[https://github.com/JungDaewook/Todo-JavaScript/pull/7/commits/efe91cb401ee7d41a10a0d9551e6de47d4aa3528](url)
위 커밋과 같이 html코드의 숨김처리 코드를 삭제하고 js 파일에서 list를 20개 이상 추가하지 못하도록 수정 완료.

JungDaewook replied:
[gh pr checkout 7](url)

JungDaewook replied:
> @JungDaewook
> 
> 해당 스타일은 아래 요구사항을 만족시키기 위해 작성된 코드로 보이는데요 🤔
> 
> * “할 일 목록"을 보여줄 수 있는 **리스트**가 존재한다.
>   
>   * 할 일 목록은 최대 20개까지 입력 가능하다.
> 
> 하지만, 해당 코드는 최대 20개 까지 입력이 가능하다가 아니라
> 
> "20개 이후의 리스트 항목을 숨김처리(`display: none`)한다"의 요구사항을 만족시키는 것 같아요.
> 
> 실제로는, 20개 이후의 항목들이 생성되고 있습니다. 수정이 필요하지 않을까요?

[답변/수정 완료]
efe91cb
위 커밋과 같이 html코드의 숨김처리 코드를 삭제하고 js 파일에서 list를 20개 이상 추가하지 못하도록 수정 완료.

JungDaewook replied:
[추가수정사한]
js로 todoItem 개수가 20개 넘지 못하도록 구현을 했다고 생각을 했는데 이제 보니 삭제를 했을 때 todoItemCount 가 1씩 감소하도록 구현하지 않아서 그 부분도 수정 후 다시 replt 남길게요

JungDaewook replied:
[추가수정사항]
js로 todoItem 개수가 20개 넘지 못하도록 구현을 했다고 생각을 했는데 이제 보니 삭제를 했을 때 todoItemCount 가 1씩 감소하도록 구현하지 않아서 그 부분도 수정 후 다시 reply 남길게요

JungDaewook replied:
[추가수정완료]
https://github.com/JungDaewook/Todo-JavaScript/pull/7/commits/7cbd623e63d77c55b99a79e38d36279e10c49784
위 커밋으로 todoItem이 삭제 될 때마다 todoItemCount가 하나씩 줄어듦으로써 removeTodoItem함수와 삭제 기능이 정상적으로 작동하도록 수정 완료.

unpo88 commented on addTodoItem.js:
```diff
@@ -1,8 +1,8 @@
-    todoList.appendChild(todoItem);
-  }
\ No newline at end of file
+  const todoInputText = document.getElementById('todoInputText').value;
+  const todoItem = document.createElement('li');
+  const textNode = document.createTextNode(todoInputText);
+  todoItem.appendChild(textNode);
+  todoList = document.getElementById('todoList');
+  todoList.appendChild(todoItem);
```
@JungDaewook 

간단한 가독성을 위해 다음과 같이 수정하면 더 좋지 않을까요?
- 개행문자 추가
- `todoInputText` -> `todoInputTextValue` 로 네이밍 수정

``` javascript
const todoInputTextValue = document.getElementById('todoInputText').value;
const todoItem = document.createElement('li');
const textNode = document.createTextNode(todoInputTextValue);

todoItem.appendChild(textNode);
todoList = document.getElementById('todoList');
todoList.appendChild(todoItem);
```

JungDaewook replied:
> @JungDaewook
> 
> 간단한 가독성을 위해 다음과 같이 수정하면 더 좋지 않을까요?
> 
> * 개행문자 추가
> * `todoInputText` -> `todoInputTextValue` 로 네이밍 수정
> 
> ```js
> const todoInputTextValue = document.getElementById('todoInputText').value;
> const todoItem = document.createElement('li');
> const textNode = document.createTextNode(todoInputTextValue);
> 
> todoItem.appendChild(textNode);
> todoList = document.getElementById('todoList');
> todoList.appendChild(todoItem);
> ```

[답변/수정완료/질문]

[답변]
[https://github.com/JungDaewook/Todo-JavaScript/pull/7/commits/0758b17c3c2cabeefe8317606e56bf445f9f1578]
위 커밋으로 수정완료하였습니다.

[질문]
혹시 'todoInputText' 에서 'todoInputTextValue' 로 바꾸는 것이 좋겠다고 한 이유가 입력값의 id인 'todoInputText' 와 헷갈릴까봐가 맞나요?

JungDaewook replied:
[https://github.com/JungDaewook/Todo-JavaScript/pull/7/commits/efe91cb401ee7d41a10a0d9551e6de47d4aa3528
](url)

JungDaewook replied:
[https://github.com/JungDaewook/Todo-JavaScript/pull/7/commits/0758b17c3c2cabeefe8317606e56bf445f9f1578](url)

JungDaewook replied:
> @JungDaewook
> 
> 간단한 가독성을 위해 다음과 같이 수정하면 더 좋지 않을까요?
> 
> * 개행문자 추가
> * `todoInputText` -> `todoInputTextValue` 로 네이밍 수정
> 
> ```js
> const todoInputTextValue = document.getElementById('todoInputText').value;
> const todoItem = document.createElement('li');
> const textNode = document.createTextNode(todoInputTextValue);
> 
> todoItem.appendChild(textNode);
> todoList = document.getElementById('todoList');
> todoList.appendChild(todoItem);
> ```

[답변/수정완료/질문]

[답변]

[답변/수정완료/질문]

[답변]
[https://github.com/https://github.com/JungDaewook/Todo-JavaScript/pull/7/commits/0758b17c3c2cabeefe8317606e56bf445f9f1578]
위 커밋으로 수정완료하였습니다.

[질문]
혹시 'todoInputText' 에서 'todoInputTextValue' 로 바꾸는 것이 좋겠다고 한 이유가 입력값의 id인 'todoInputText' 와 헷갈릴까봐가 맞나요?
위 커밋으로 수정완료하였습니다.

[질문]
혹시 'todoInputText' 에서 'todoInputTextValue' 로 바꾸는 것이 좋겠다고 한 이유가 입력값의 id인 'todoInputText' 와 헷갈릴까봐가 맞나요?

JungDaewook replied:
a116e53

JungDaewook replied:
> @JungDaewook
> 
> 간단한 가독성을 위해 다음과 같이 수정하면 더 좋지 않을까요?
> 
> * 개행문자 추가
> * `todoInputText` -> `todoInputTextValue` 로 네이밍 수정
> 
> ```js
> const todoInputTextValue = document.getElementById('todoInputText').value;
> const todoItem = document.createElement('li');
> const textNode = document.createTextNode(todoInputTextValue);
> 
> todoItem.appendChild(textNode);
> todoList = document.getElementById('todoList');
> todoList.appendChild(todoItem);
> ```

[답변/수정완료/질문]

[답변] 
0758b17
 위 커밋으로 수정완료하였습니다.

[질문] 
혹시 'todoInputText' 에서 'todoInputTextValue' 로 바꾸는 것이 좋겠다고 한 이유가 입력값의 id인 'todoInputText' 와 헷갈릴까봐가 맞나요?

unpo88 replied:
@JungDaewook 

맞습니다~

unpo88 reviewed (changes requested):
@JungDaewook 

수정이 필요한 사항에 대해서 코멘트 남겨놓았습니다.

------------------------

**[추가 과제]**
아래 영상을 살펴보면 `input` 에 값을 입력하고, 추가를 눌렀을 때, `input` 값이 초기화되지 않고, 그대로 유지되는 경우가 있어요. 요것을 초기화해주려면 어떻게 해야할까요? 🤔 

https://user-images.githubusercontent.com/14287771/161287161-1b4eb952-4607-4449-a287-2024fe9a3270.mov

Commit:
modified the first and second comment
1. DeleteTodoItem이라는 변수명을 deleteTodoItem으로 변경 (lowerCamelCase)
1-1. todoItemCount라는 변수명을 TodoItemCount(UpperCamelCase)로 변경 전역 변수라 지역 변수와 구분이 필요할 것이라고 생각하여 변경함.
2. 가독성을 위하여 addTodoItem이라는 함수에서 변수선언부분과 todoItem에 입력받은 텍스트를 추가해주는 부분, todoList에 todoItem을 추가해주는 부분을 가독성을 위해 줄바꿈으로 구분.

Commit:
코멘트에 따라 todoInputText -> todoInputTextValue 로 변경

Commit:
코멘트를 반영하여 20개를 초과하여 todoItem이 추가되지 못하도록 구현
1. html파일에서 실제 todoItem이 추가되더라도 안보여주는 것을 js파일에서 실제로 todoItem이 추가되지 못하도록하고 html파일에 있던 코드는 삭제
2. TodoItemCount가 초기에 0이 되어야 1개가 추가될 때마다 todoItem의 개수를 나타내므로 TodoItemCount의 초기값을 0으로 설정

Commit:
입력창에 아무것도 입력하지 않아도 list에 추가가 되는 문제 해결

JungDaewook commented:
[수정]

[https://github.com/JungDaewook/Todo-JavaScript/pull/7/files/efe91cb401ee7d41a10a0d9551e6de47d4aa3528..bd03bb5fde7778eee5163f137223e8788b2f7a1d](url)
입력창에 아무것도 입력하지 않아도 입력 버튼을 눌렀을 때 추가가 되는 문제 해결

Commit:
추가 과제인 값을 입력했을 때 입력창이 초기화되는 부분 해결

JungDaewook commented:
> @JungDaewook
> 
> 수정이 필요한 사항에 대해서 코멘트 남겨놓았습니다.
> 
> **[추가 과제]** 아래 영상을 살펴보면 `input` 에 값을 입력하고, 추가를 눌렀을 때, `input` 값이 초기화되지 않고, 그대로 유지되는 경우가 있어요. 요것을 초기화해주려면 어떻게 해야할까요? 🤔
> 
>  2022-04-01.11.46.52.mov

[과제 달성]

[https://github.com/JungDaewook/Todo-JavaScript/pull/7/commits/a116e53f2e01fb2e248a08ebf87081818f2178d4](url)
위 커밋과 같이 입력값을 입력했을 때 입력창이 초기화되도록 구현.

JungDaewook commented:
[수정]

https://github.com/JungDaewook/Todo-JavaScript/pull/7/commits/a116e53f2e01fb2e248a08ebf87081818f2178d4

JungDaewook commented:
[수정]

https://github.com/JungDaewook/Todo-JavaScript/pull/7/commits/a116e53f2e01fb2e248a08ebf87081818f2178d4
위 커밋으로 같이 입력창에 아무것도 입력하지 않았을 때 list에 추가가 되는 문제 해결

JungDaewook commented:
> @JungDaewook
> 
> 수정이 필요한 사항에 대해서 코멘트 남겨놓았습니다.
> 
> **[추가 과제]** 아래 영상을 살펴보면 `input` 에 값을 입력하고, 추가를 눌렀을 때, `input` 값이 초기화되지 않고, 그대로 유지되는 경우가 있어요. 요것을 초기화해주려면 어떻게 해야할까요? 🤔
> 
>  2022-04-01.11.46.52.mov

[과제 달성]

https://github.com/JungDaewook/Todo-JavaScript/pull/7/commits/a116e53f2e01fb2e248a08ebf87081818f2178d4
위 커밋으로 입력값을 입력했을 때 입력창이 초기화되는 기능 구현.

JungDaewook commented:
[수정]

bd03bb5
위 커밋을 통해 입력창에 아무것도 없을 때 값이 입력되지 않게 수정.

JungDaewook commented:
https://github.com/JungDaewook/Todo-JavaScript/pull/7/commits/efe91cb401ee7d41a10a0d9551e6de47d4aa3528#diff-191efbd1925221a4ac2fb2e430714b9072551a60c97d771637ca849ece21a626R1

JungDaewook committed:
Merge branch 'main' into modifyThatImissed

unpo88 commented on functionsOfTodoList.js:
```diff
@@ -0,0 +1,30 @@
+  const todoItem = document.createElement('li');
+  const textNode = document.createTextNode(todoInputTextValue);
+
+  if (TodoItemCount >=20){
+    return ;
+  }
```
@JungDaewook 

[적극 제안]

20과 같은 상수는 다음과 같이 상수로 선언해주는 것이 의미적으로 어떤 것을 하는지 이해하기 좋지 않을까요?

그리고 다음과 같이 해당 조건을 변수로 나타내면 해당 행위를 이해하는데 조금 더 이해하기 쉽지 않을까요?

``` javascript
const MAX_TODO_ITEM_COUNT = 20

...

function addTodoItem() {
    const isMaximumTodoItemCount = TodoItemCount >= MAX_TODO_ITEM_COUNT
    ... 
    if (isMaximumTodoItemCount ) {
        ...
    }
```

추가로 TodoItemCount 역시 상수이기 때문에

TODO_ITEM_COUNT와 같은 네이밍 규칙을 가져가는것이 어떠한가요?

JungDaewook replied:
> @JungDaewook
> 
> [적극 제안]
> 
> 20과 같은 상수는 다음과 같이 상수로 선언해주는 것이 의미적으로 어떤 것을 하는지 이해하기 좋지 않을까요?
> 
> 그리고 다음과 같이 해당 조건을 변수로 나타내면 해당 행위를 이해하는데 조금 더 이해하기 쉽지 않을까요?
> 
> ```js
> const MAX_TODO_ITEM_COUNT = 20
> 
> ...
> 
> function addTodoItem() {
>     const isMaximumTodoItemCount = TodoItemCount >= MAX_TODO_ITEM_COUNT
>     ... 
>     if (isMaximumTodoItemCount ) {
>         ...
>     }
> ```
> 
> 추가로 TodoItemCount 역시 상수이기 때문에
> 
> TODO_ITEM_COUNT와 같은 네이밍 규칙을 가져가는것이 어떠한가요?

[제안 수용/리팩터링 브랜치 신규 생성 후 수정 예정]
리팩터링 관련된 부분으로 보이는데 제안을 수용하여 수정할 새로운 리팩터링 브랜치를 만들어서 수정하도록 할게요!

unpo88 commented on functionsOfTodoList.js:
```diff
@@ -0,0 +1,30 @@
+    return ;
+  }
+
+  if(todoInputTextValue === ''){
+    return;
+  }
```
@JungDaewook

[적극/제안]

여기도 마찬가지로 다음과 같이 행위로 표현해주는건 어떠한가요?

``` javascript
const isEmptyTodoInputTextValue = todoInputTextValue === ''

if (isEmptyTodoInputTextValue) {
    return ;
}
```

JungDaewook replied:
> @JungDaewook
> 
> [적극/제안]
> 
> 여기도 마찬가지로 다음과 같이 행위로 표현해주는건 어떠한가요?
> 
> ```js
> const isEmptyTodoInputTextValue = todoInputTextValue === ''
> 
> if (isEmptyTodoInputTextValue) {
>     return ;
> }
> ```

[제안 수용/리팩터링 브랜치 신규 생성 후 수정 예정]
리팩터링 관련된 부분으로 보이는데 제안을 수용하여 수정할 새로운 리팩터링 브랜치를 만들어서 수정하도록 할게요!

unpo88 reviewed (changes requested):
@JungDaewook 

코멘트 남겨놓았고

@JungDaewook 

[적극/제안]

추가로 UX/UI 상, 20개가 넘었다거나, 빈 Text를 추가했을 때 그냥 추가가 안되면

사용자가 왜 추가가 안되는지 인지하기 어렵지 않을까요?

사용자가 인지하기 쉽게 하려면 어떻게 해야할까요?

JungDaewook commented:
> @JungDaewook
> 
> 코멘트 남겨놓았고
> 
> @JungDaewook
> 
> [적극/제안]
> 
> 추가로 UX/UI 상, 20개가 넘었다거나, 빈 Text를 추가했을 때 그냥 추가가 안되면
> 
> 사용자가 왜 추가가 안되는지 인지하기 어렵지 않을까요?
> 
> 사용자가 인지하기 쉽게 하려면 어떻게 해야할까요?

[제안 수용/리팩터링 브랜치 신규 생성 후 수정 예정]
리팩터링 관련된 부분으로 보이는데 제안을 수용하여 수정할 새로운 리팩터링 브랜치를 만들어서 수정하도록 할게요! 위 코멘트 두 개를 수정하면서 같이 반영하여 수정하도록 할게요!

Commit:
리팩터링 브랜치 생성을 위한 커밋
TodoItemCount를 감소시키려고 여러가지 수정을 하다가 그것을 기존 상태로 변경하여 수정사항이 생겼다고 인식이 되었고 그것을 커밋하고 새로운 브랜치를 생성하기 위해 커밋을 해준다.

Commit:
addTodoItem()에서 TodoInputTextValue라는 변수에 입력값을 받아오는 방식에서 입력값을 받아오는 함수를 따로 만들어서 입력값을 받아오도록 수정.

Commit:
코멘트에 따라 MAX_TODO_ITEM_COUNT, EMPTY_TODO_ITEM_COUNT 변수 생성, TodoItemCount에서 TODO_ITEM_COUNT로 변수명 변경, isEmptyTodoInputText(), isBiggerThanMAX_TODO_ITEM_COUNT 함수 생성
1.MAX_TODO_ITEM_COUNT, EMPTY_TODO_ITEM_COUNT 변수 생성
2. TodoItemCount -> TODO_ITEM_COUNT 로 변수명 변경
3. srp 원칙에 의거, isEmptyTodoInputText(), isBiggerThanMAX_TODO_ITEM_COUNT() 생성
4. addTodoItem()에서 isEmptyTodoInputText()를 isBiggerThanMAX_TODO_ITEM_COUNT() 보다 먼저 오도록 변경(빈 입력값이 들어가있는지 확인해주는게 우선순위가 높다고 판단했기 때문)

Commit:
스터디 때 했던 피드백에 따라 addTodoItem() 함수 안에서 타입체크를 하는 것이 아니라 handleAddButton() 함수를 만듦으로써 타입체크를 먼저 한 후에 addTodoItem()을 실행하는 방식으로 변경
1. handleAddButton() 생성, addTodoItem() 안의 타입체크 함수들 삭제.
2. EMPTY_TODO_ITEM -> EMPTY_TODO_INPUT_TEXT로 변경 이전 커밋에서 변수명 설정을 잘못하여 변경.
3. isEmptyTodoInputText() 함수와 isBiggerThanMAX_TODO_ITEM_COUNT() 함수의 선언 순서를 변경 (isEmptyTodoInputText()가 먼저 실행되기 때문.)

Commit:
의도치 않게 생성된 파일 제거, 주석처리 제거
1. addTodoItem.js, .vscode/launch.json 파일 제거
2. 타입체크 주석처리 제거

Commit:
코멘트에 의해 todoItemCount로 변수명 다시 변경

Commit:
EMPTY_TODO_ITEM 변수 삭제

Commit:
isBiggerThanMaxTodoItemCount로 함수명 변경

Commit:
maxlength 다시 생성

Commit:
코멘트에 따라 isEmptyTodoInputText(), isBiggerThanTodoItemcount(), handleAddbutton() 수정 완료.

Commit:
refactorin1의 코멘트에 의해 removeTodoItem(todoItemId)로 수정 완료.

Commit:
코멘트에 의해 타입체크 함수 두개의 if문을 한줄로 간소화

JungDaewook committed:
Merge branch 'modifyThatImissed' into refactoring1

Commit:
todoItem을 삭제할 때 todoItemCount가 하나씩 줄어들도록 구현

Commit:
addTodoItem.js 삭제

JungDaewook commented:
https://github.com/JungDaewook/Todo-JavaScript/pull/10#discussion_r852196496
https://github.com/JungDaewook/Todo-JavaScript/pull/10#discussion_r862473140
https://github.com/JungDaewook/Todo-JavaScript/pull/10#discussion_r862473200
refactoring1에서 다른 코멘트는 다 처리하였고 위의 세가지 코멘트만 확인해주시면 될 것 같아요. modifyThatImissed에서의 코멘트도 다 처리한 것 같은데 코멘트 남겨주실 부분이 있으면 더 남겨주시고 아니라면 merge하고 전체삭제 기능을 구현하면 될 것 같아요.

unpo88 commented on functionsOfTodoList.js:
```diff
@@ -0,0 +1,50 @@
+}
+
+function handleAddButton(){
+  if(isEmptyTodoInputText()) {
+    alert("입력창에 할 일을 입력하십시오.");
+  }
+
+  if(isBiggerThanMaxTodoItemCount()){
+    alert("할 일은 20개까지만 입력할 수 있습니다.");
+  }
+
+  if(!isEmptyTodoInputText() && !isBiggerThanMaxTodoItemCount()){
+    addTodoItem();
+  }
```
@JungDaewook 

[필수]

`isEmptyTodoInputText`, `isBiggerThanMaxTodoItemCount` 는 `handleAddButton`에서만 사용되는 것으로 보이는데 맞을까요?

그렇다면, 응집도를 높이기 위해

[Comment](https://github.com/JungDaewook/Todo-JavaScript/pull/7#discussion_r851629218)에 남겨놓은 것처럼 함수가 아닌 변수로 선언하는것은 어떠한가요~?

해당 코멘트 반영하고 Merge 부탁드립니다 🙇

JungDaewook replied:
[반영 완료/merge하기 전 최종 확인]
https://github.com/JungDaewook/Todo-JavaScript/pull/7/commits/f6d5be36996158e89519a95e5e79ca67037058d8
위 커밋으로 반영완료했는데 제가 원섭님의 의도대로 한 것이 맞는지 확인받고 싶어요.

unpo88 replied:
@JungDaewook 

맞습니다~ 👍

unpo88 reviewed (approved):
@JungDaewook 

코멘트 하나 추가로 남겨놓았습니다~

확인 부탁드립니다 🙏

Commit:
코멘트에 따라 타입체크 함수를 handleAddButton 함수 안의 변수로 설정하여 응집도를 높임.

unpo88 reviewed (approved):
LGTM
